lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -785,6 +785,15 @@ static Optional<RequirementMatch> findMissingGenericRequirementForSolutionFix(
     return Optional<RequirementMatch>();
   }
 
+  type = solution.simplifyType(type);
+  missingType = solution.simplifyType(missingType);
+
+  missingType = missingType->mapTypeOutOfContext();
```

Comment:
Are you still seeing type variables after this substitution? The types in these requirements should already be mapped into the appropriate context, so the conditions below don't make sense to me.

Reply:
The types start out in the context of the type we're trying to find witnesses for and thus can contain `generic_type_param_type` and would emit as e.g. `CountSteps1<τ_0_0>.Index`. Mapping them out and into the context of the conformance transforms them into containing `primary_archetype_type` in the equivalent places so we emit e.g. `CountSteps1<T>.Index`.

Reply:
This part is what SR-12789 (the resolved is-duplicated-by bug) was complaining about.